packages/back-end/src/scim/groups/patchGroup.ts

```diff
@@ -93,11 +95,26 @@ export async function patchGroup(
         }
       } else if (op === "replace" && !path) {
         // Update Group object
-        await updateTeamMetadata(team.id, org.id, {
-          ...team,
-          name: (value as BasicScimGroup).displayName,
-          managedByIdp: true,
-        });
+        const updatedTeam = cloneDeep(team);
+
+        updatedTeam.name = (value as BasicScimGroup).displayName;
+        updatedTeam.managedByIdp = true;
+
+        const growthbookRole = (value as BasicScimGroup).growthbookRole;
+
+        if (growthbookRole && growthbookRole !== team.role) {
+          if (!isRoleValid(growthbookRole)) {
+            return res.status(400).json({
+              schemas: ["urn:ietf:params:scim:api:messages:2.0:Error"],
+              status: "400",
+              detail: `"${growthbookRole}" is not a valid GrowthBook role.`,
+            });
+          }
+
+          updatedTeam.role = growthbookRole;
+        }
+
+        await updateTeamMetadata(team.id, org.id, updatedTeam);
```

Comment:
I think you can replace this with 
```
        await updateTeamMetadata(team.id, org.id, {
          ...team,
          name: (value as BasicScimGroup).displayName,
          managedByIdp: true,
          growthbookRole,
        });
```
and have the same effect without needing `cloneDeep`. If its undefined it just wont be added and if its not a valid role we'll return the error before we get to update

Reply:
Ah, yea - much cleaner. Thanks!